packages/ui/src/containers/PhoneForm/PhoneForm.tsx

```diff
@@ -0,0 +1,128 @@
+import classNames from 'classnames';
+import { useCallback, useEffect, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import Button from '@/components/Button';
+import ErrorMessage from '@/components/ErrorMessage';
+import { PhoneInput } from '@/components/Input';
+import PasswordlessSwitch from '@/containers/PasswordlessSwitch';
+import TermsOfUse from '@/containers/TermsOfUse';
+import useForm from '@/hooks/use-form';
+import usePhoneNumber from '@/hooks/use-phone-number';
+import useTerms from '@/hooks/use-terms';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  className?: string;
+  // eslint-disable-next-line react/boolean-prop-naming
+  autoFocus?: boolean;
+  hasTerms?: boolean;
+  hasSwitch?: boolean;
+  errorMessage?: string;
+  clearErrorMessage?: () => void;
+  onSubmit: (phone: string) => Promise<void>;
+};
+
+type FieldState = {
+  phone: string;
+};
+
+const defaultState: FieldState = { phone: '' };
+
+const PhoneForm = ({
+  autoFocus,
+  hasTerms = true,
+  hasSwitch = false,
+  className,
+  errorMessage,
+  clearErrorMessage,
+  onSubmit,
+}: Props) => {
+  const { t } = useTranslation();
+
+  const { termsValidation } = useTerms();
+  const { countryList, phoneNumber, setPhoneNumber, isValidPhoneNumber } = usePhoneNumber();
+
+  const { fieldValue, setFieldValue, validateForm, register } = useForm(defaultState);
+
+  /*  Clear the form error when input field is updated */
+  const errorMessageRef = useRef(errorMessage);
```

Comment:
I'm wondering why do we have to use a ref to deal with the error message

Reply:
updated